kmir/src/kmir/smir.py

```diff
@@ -79,7 +79,17 @@ def function_arguments(self) -> dict[str, list[dict]]:
     @cached_property
     def function_symbols(self) -> dict[int, dict]:
         fnc_symbols = {ty: sym for ty, sym, *_ in self._smir['functions'] if type(ty) is int}
+        # by convention, Ty -1 is used for 'main' if it exists
         fnc_symbols[-1] = {'NormalSym': self.main_symbol}
+
+        # function items not present in the SMIR lookup table are added with negative Ty ID
+        missing = [name for name in self.items.keys() if {'NormalSym': name} not in fnc_symbols.values()]
+
+        fake_ty = -2
+        for name in missing:
+            fnc_symbols[fake_ty] = {'NormalSym': name}
+            fake_ty -= 1
+
```

Comment:
```suggestion
        # by convention, Ty -1 is used for 'main' if it exists
        # function items not present in the SMIR lookup table are added with negative Ty ID
        missing = ['main'] + [name for name in self.items.keys() if {'NormalSym': name} not in fnc_symbols.values()]
        fake_ty = -1
        for name in missing:
            fnc_symbols[fake_ty] = {'NormalSym': name}
            fake_ty -= 1

```

Just a thought.

Reply:
Not sure, because then we may get `main` twice.
I was going to remove the `main == -1` special case but the view/show output would change